### ISSUE TITLE
Add function to create Graphviz visualizations of layers

### ISSUE THREAD

**Issue by hlin117** [Lasagne/Lasagne#638]
Similar to [keras functionality](http://keras.io/visualization/), it would be nice if there were a function to show the architecture of a neural network. 

For example, this is a network I built in Keras a few nights ago:

![screenshot 2016-03-11 00 05 27](https://cloud.githubusercontent.com/assets/1865885/13727485/c6f10e50-e8b9-11e5-9525-4870be00faf5.png)

Related: 
* #106 (though this is focusing on visualizing what the layers learn, not the shapes of the layers)
* #597 (I'm referencing this issue because this function can also facilitate a debugging process)

**Comment by benanne** [comment]
This would be great to have, but I think it falls outside the scope of the main library a bit (and would require extra dependencies), so I think it's more appropriate as a recipe: https://github.com/Lasagne/Recipes

We already have something in there that generates a textual representation, but something graphical would be nice as well :)

**Comment by hlin117** [comment]
Good point about the dependencies. That's something we definitely want to
avoid.

Out of curiosity, what is the function that generates the textual
representation?

**Comment by benanne** [comment]
Sorry, I should have linked to that directly! Here it is: https://github.com/Lasagne/Recipes/blob/master/utils/network_repr.py

**Comment by ebenolson** [comment]
I made something along those lines a while back, although it probably needs updating to work with the current code: https://gist.github.com/ebenolson/1682625dc9823e27d771

**Comment by Sandy4321** [comment]
Cool, data science is visualisation, do not it?
